api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -617,15 +617,14 @@ public static void applyLogLevel(String logClass, String logLevel) {
 		}
 		return ret;
 	}
-	
+
 	public static <Arg1, Arg2 extends Arg1> boolean nullSafeEquals(Arg1 d1, Arg2 d2) {
 		if (d1 == null) {
 			return d2 == null;
 		} else if (d2 == null) {
 			return false;
-		} else {
-			return d1.equals(d2);
 		}
+		return (d1 instanceof Timestamp || d2 instanceof Timestamp) ? compare((Date) d1, (Date) d2) == 0 : d1.equals(d2);
```

Comment:
Wouldn't it actually be better to check if instance of Date and then delegate to compare(Date, Date)?